CMakeLists.txt

```diff
@@ -68,6 +68,8 @@ endif()
 
 add_subdirectory(src)
 
+add_subdirectory(test)
```

Comment:
Should we put this under just `example`. Fpm can pick up demo programs and run them as well, means we have to deploy this directory with fpm (maybe for a follow-up PR).